src/OrleansProviders/Streams/LoadShedQueueFlowController.cs

```diff
@@ -24,7 +24,7 @@ public class LoadShedQueueFlowController : IQueueFlowController
         /// <param name="options">The silo satistics options.</param>
         /// <param name="percentOfSiloSheddingLimit">Percentage of load shed limit which triggers a reduction of queue read rate.</param>
         /// <returns></returns>
-        public static IQueueFlowController CreateAsPercentOfLoadSheddingLimit(SiloStatisticsOptions options, int percentOfSiloSheddingLimit = SiloStatisticsOptions.DEFAULT_LOAD_SHEDDING_LIMIT)
+        public static IQueueFlowController CreateAsPercentOfLoadSheddingLimit(LoadSheddingOptions options, int percentOfSiloSheddingLimit = LoadSheddingOptions.DEFAULT_LOAD_SHEDDING_LIMIT)
         {
             if (percentOfSiloSheddingLimit < 0.0 || percentOfSiloSheddingLimit > 100.0) throw new ArgumentOutOfRangeException(nameof(percentOfSiloSheddingLimit), "Percent value must be between 0-100");
```

Comment:
isn't the logic here is a bug and we are going to remove it ?

Reply:
I discussed it with Jason. It's not a bug. It's a protection from a corner case of stream pulling agents potentially overloading silo and preventing grain calls from getting processed.